service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/handlers/C2UploadedEventHandlerEmailTemplateTest.java

```diff
@@ -0,0 +1,93 @@
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import uk.gov.hmcts.reform.fpl.events.C2UploadedEvent;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.common.C2DocumentBundle;
+import uk.gov.hmcts.reform.fpl.model.notify.c2uploaded.C2UploadedTemplate;
+import uk.gov.hmcts.reform.fpl.request.RequestData;
+import uk.gov.hmcts.reform.fpl.service.email.NotificationService;
+import uk.gov.hmcts.reform.fpl.service.email.content.C2UploadedEmailContentProvider;
+import uk.gov.hmcts.reform.fpl.testingsupport.email.EmailTemplateTest;
+import uk.gov.hmcts.reform.idam.client.IdamClient;
+import uk.gov.hmcts.reform.idam.client.models.UserInfo;
+
+import static org.mockito.BDDMockito.given;
+
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.CASE_TYPE;
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.JURISDICTION;
+import static uk.gov.hmcts.reform.fpl.enums.UserRole.LOCAL_AUTHORITY;
+import static uk.gov.hmcts.reform.fpl.handlers.NotificationEventHandlerTestData.AUTH_TOKEN;
+import static uk.gov.hmcts.reform.fpl.testingsupport.email.EmailContent.emailContent;
+import static uk.gov.hmcts.reform.fpl.testingsupport.email.SendEmailResponseAssert.assertThat;
+import static uk.gov.hmcts.reform.fpl.utils.CoreCaseDataStoreLoader.caseData;
+
+@SpringBootTest(classes = {
+    C2UploadedEventHandler.class,
+    NotificationService.class,
+    ObjectMapper.class
+})
+
+public class C2UploadedEventHandlerEmailTemplateTest extends EmailTemplateTest {
+    private final C2DocumentBundle c2DocumentBundle = C2DocumentBundle.builder().build();
+    private final String respondentLastName = "Smith";
+    private final String calloutText = "Smith, SACCCCCCCC5676576567";
+    private final String caseUrl = "null/case/" + JURISDICTION + "/" + CASE_TYPE + "/12345#C2Tab";
+
+    @Autowired
+    private C2UploadedEventHandler underTest;
+
+    @MockBean
+    private IdamClient idamClient;
+
+    @MockBean
+    private RequestData requestData;
+
+    @MockBean
+    private C2UploadedEmailContentProvider c2UploadedEmailContentProvider;
+
+    @Test
+    void notifyAdmin() {
+        CaseData caseData = caseData();
+
+        given(requestData.authorisation()).willReturn(AUTH_TOKEN);
+
+        given(idamClient.getUserInfo(AUTH_TOKEN)).willReturn(
+            UserInfo.builder().sub("hmcts-non-admin@test.com").roles(LOCAL_AUTHORITY.getRoles()).build());
+
+        given(c2UploadedEmailContentProvider.getNotifyData(caseData,
+            c2DocumentBundle.getDocument()))
+            .willReturn(
+                C2UploadedTemplate.builder()
+                    .callout(calloutText)
+                    .respondentLastName(respondentLastName)
+                    .caseUrl(caseUrl)
+                    .documentLink(DOCUMENT_LINK)
```

Comment:
so this document link is not actually a Map<String, Object> in this situation, it's a string - essentially it's another URL. We have two ways of 'sending a document': one is to attach it by doing what you're doing here with the JSONObject, the other way is to send a case link to that document (so another url, but one that links directly to the document in the case - and will require user to be logged in/signed in). We want to use the latter as much as possible because there were some GDPR concerns about using the attached document process (anyone with this link can download the document, it was frowned upon) - so wherever we can (i.e, if user has case access), we will give them a URL linking to the document - if they dont have case access, then we can use the JSONObject approach. You havent changed the main code but just need to update this part of the test to reflect what happens in the live template

Reply:
you'll see what I mean if you look into C2UploadedEventHandler -> C2UploadedEmailContentProvider, the getDocumentUrl method is just building a String (and you'll see above it the method for the other approach, which is to 'attach' the document)